calamares: 3.3.13 -> 3.4.0, qt6-ize, clean up

K900 opened the pull request:
- vendor calamares-nixos-extensions
- remove all the /run hardcoding
- remove run as non-root bullshit
- drop patches:
  - 0002 - replaced with wrapper args
  - 0003 - no, we're not doing this downstream
  - 0004 - replaced with custom page + tiny patch
  - 0005 - too much effort
  - 0007 - upstream fixed this better
  - 0008 - we can just set it in the config instead

TODO:
- [x] does this actually fucking work
- [x] make commit history pretty
- [x] make unfree page look tolerable
- [x] late wrap calamares with extensions
- [x] fix pkexec nonsense
- die

TODO (later):
- [ ] figure out what the luks patch is for
- [ ] upstream patches???


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

adriaandegroot committed:
Docs: mention the specific module examples

ShalokShalom committed:
Fix broken link

Probably happened due to copy/paste mistake or something

adriaandegroot committed:
Merge pull request #1 from ShalokShalom/patch-1

[kaos branding] Fix broken link

ollieparanoid committed:
[mobile] add mobile friendly module and branding

Create a new module and branding targeted at the smartphone and tablet
use case. The installation process is significantly different from your
typical PC install:
* The UI must fit mobile screens, the amount of inputs visible on one
  screen must be reduced.
* Every input must be controllable with an on screen keyboard.
* Full disk encryption is done with GRUB in Calamares, but mobile
  devices use other bootloaders (u-boot, lk).
* Complex partitioning screens are not useful for most users, as the
  expectation is not that multiple OS can be installed next to each
  other.

It would be nice if the regular Calamares modules worked with mobile
devices eventually. But this can't be done in one step (and before that,
the QML <> QWidgets related refactoring should probably be completed?),
therefore let's add it to calamares-extensions first.

This initial version is far from perfect, but what is there works
reliably. It should give a good shared base to iterate upon.

Implemented screens:
* Welcome
* Lockscreen PIN
  Set the user password, but as numeric PIN. Because that is what the
  current generation of mobile friendly Linux DEs expects (Plasma Mobile
  and Phosh).
* SSH server: enable or disable
  To debug Linux distributions on phones, SSH is essential. Give
  advanced users the opportunity to enable it and casual users the
  option to skip this.
* Credentials for SSH server
  The SSH server has a dedicated user account with a stronger password.
  Users are encouraged to replace password auth with SSH key auth after
  the installation.
* Full disk encryption: enable or disable
* Full disk encryption password

Screenshots:
https://postmarketos.org/blog/2020/07/15/pinephone-ce-preorder/#installer-with-optional-ssh-server-and-encryption

There is no partitioning screen. The target partition, which will be
formatted and optionally encrypted is defined in mobile.conf as
targetDeviceRoot. It is expected that the installer OS will use this to
install to an empty partition on the same medium as the installer (SD
card or eMMC) as explained here:
https://wiki.postmarketos.org/wiki/On-device_installer

I ended up throwing everything into one module instead of several,
because of a Qt Virtual Keyboard bug:
https://gitlab.com/postmarketOS/postmarketos-ondev/-/issues/18

Based on, but adjusted to be distro-agnostic:
https://gitlab.com/postmarketOS/postmarketos-ondev/-/releases/0.2.1

adriaandegroot committed:
Merge pull request #3 from ollieparanoid/calamares-mobile

[mobile] add mobile friendly module and branding

ollieparanoid committed:
[mobile] use SPDX-FileCopyrightText

ollieparanoid committed:
CMakeLists: add find_package YAMLCPP

Find yamlcpp, even if it is named libyaml-cpp.so (with a dash) as it is
the case in Alpine Linux.

adriaandegroot committed:
Merge pull request #4 from ollieparanoid/calamares-mobile

[mobile] use SPDX-FileCopyrightText

adriaandegroot committed:
Merge pull request #5 from ollieparanoid/fix-cmake-yamlcpp

CMakeLists: add find_package YAMLCPP

adriaandegroot committed:
[mobile] Conventionally, create null QString with default constructor

adriaandegroot committed:
[mobile] QPair with const member types is problematic

Depending on compiler version and Qt version, this may or may not
compile; typical errors look like this:

/usr/include/qt5/QtCore/qlist.h:471:35: error: use of deleted function 'QPair<const QStringList, const QString>& QPair<const QStringList, const QString>::operator=(const QPair<const QStringList, const QString>&)'
  471 |     else *reinterpret_cast<T*>(n) = t;
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~
/usr/include/qt5/QtCore/qpair.h:49:8: note: 'QPair<const QStringList, const QString>& QPair<const QStringList, const QString>::operator=(const QPair<const QStringList, const QString>&)' is implicitly deleted because the default definition would be ill-formed:
   49 | struct QPair
      |        ^~~~~

adriaandegroot committed:
[mobile] Fix build (and conventions)

adriaandegroot committed:
[mobile] Fix link

There's no need to name calamaresui -- viewmodules get that
automatically -- and if it *did* need to be named, it would
be Calamares::calamaresui

adriaandegroot committed:
Add formatting guides / coding style

adriaandegroot committed:
Apply coding-style

adriaandegroot committed:
Apply coding-style

adriaandegroot committed:
Apply coding-style

adriaandegroot committed:
CMake: add version target for release script

adriaandegroot committed:
CMake: add -C to pass extra arguments to release-cmake

adriaandegroot committed:
[mobile] Missing

ollieparanoid committed:
[mobile] Make sshd feature optional

Add featureSshd config key to hide related UI screens and skip related
logic in UsersJob.

ollieparanoid committed:
[mobile] Refactor screen navigation logic

Organize the screens into features and replace all navTo() calls with
hardcoded screen names with new navNext() and navNextFeature()
functions.

ollieparanoid committed:
[mobile] navTo: add log message for screen change

ollieparanoid committed:
[mobile] navNextFeature: support skipping features

Skip whole features if a featureSshd (example for feature "sshd") or
smilar config key exists and is set to false.

ollieparanoid committed:
[mobile] Offer to install from SD to eMMC

Make it possible to install from external storage to internal storage,
if the installer was booted from the external storage medium.

ollieparanoid committed:
[mobile] skipDisabledFeatures: add skip by func

Make it possible to skip features by defining a skip function.

ollieparanoid committed:
[mobile] mobile.qml: fix whitespace error

ollieparanoid committed:
[mobile] Config.h: tweak signal related comments

Fix a typo and make the "booleans we don't read from QML don't need
a signal" message generic with one example, but not with a listing of
all these booleans. Then we won't need to adjust this comment with each
new one.

ollieparanoid committed:
[mobile] qml: use generic names for id fields

Rename "enableButton" to "firstButton" and "welcomeText" to "mainText".

adriaandegroot committed:
Merge pull request #6 from ollieparanoid/sshd-optional

[mobile] make sshd feature optional and related navigation refactoring

ollieparanoid committed:
[mobile] mobile.conf: tweak line ending

ollieparanoid committed:
[mobile] create Users job in jobs()

Don't create the user's job in MobileQmlViewStep::onLeave() anymore.
Instead, have Config::createJobs() and call that from
MobileQmlViewStep::jobs(), as suggested by Adriaan, and that's also how
it's done in the keyboard module in calamares.git. This allows getting
rid of m_jobs, and makes it less awkward to pass parameters to the job.

Related: https://github.com/calamares/calamares-extensions/pull/6#discussion_r545048454

ollieparanoid committed:
[mobile] MobileQmlViewStep: remove unused include

ollieparanoid committed:
[mobile] Don't ignore partition job errors

Stop the installation if the partition job does not finish successfully
instead of carrying on and failing later on. While at it, elaborate some
more on why it is necessary to call the partition job early.

Related: https://github.com/calamares/calamares/issues/1600

ollieparanoid committed:
[mobile] Config: remove redundant get{Bool,Str}

Use getBool and getString from utils/Variant.h instead. I had
implemented my own versions earlier, because I didn't know about the
existing ones.

ollieparanoid committed:
[mobile] *Job: 'const QString&' for args

Related: https://github.com/calamares/calamares-extensions/pull/6#discussion_r545049605

adriaandegroot committed:
Merge pull request #7 from ollieparanoid/sd-to-emmc

[mobile] Offer to install from SD to eMMC

adriaandegroot committed:
CMake: behave a little more modern

- This copies in some more settings from Calamares-proper

adriaandegroot committed:
Changes: pre-release housekeeping

adriaandegroot committed:
Changes: post-release housekeeping

Commit:
[mobile] Allow user to configure filesystem type

The user can now configure the target root filesystem type. Options
provided are ext4 (default), f2fs and btrfs.

OS Maintainers are required to provide their own BTRFS install script
which configures default submodules as required.

Allow disabling selection of fstype
This setting is disabled by default and defaults to ext4. Enabling it
will provide the user with a choice of ext4, f2fs and btrfs for the root
filesystem.

Add FS specific mkfs configuration options
This allows distribution maintainers to set the command/script used for
configuring the root filesystem on a per-fs level. Two examples of its
use are:
* F2FS: mkfs.f2fs uses -l for label, rather than -L like other filesystems.
* BTRFS: Distributions can provide a script which will configure submodules
         as requiref for snapshots to work correctly.

Use of QStringList rather than a Model:
We don't need the complexities of a full model here for a list of strings
and labelModel was the wrong one anyway. At least going down that path means
the back and forward buttons work.

ollieparanoid committed:
[mobile] wait screen: reword 20s -> 2 min

Change the message in the wait screen from "This may take up to 20
seconds" to "This may take up to two minutes". That's what Mobian needs,
and with "up to" in the sentence, it's not wrong, even if the actual
time is much shorter.

Creating a config option was considered (#10), but would only add
technical debt since the wait screen is already a hack (as described in
the big comment in runPartitionJobThenLeave()).

adriaandegroot committed:
Merge pull request #11 from ollieparanoid/wait-2min

[mobile] wait screen: reword 20s -> 2 min

adriaandegroot committed:
CI: try to run CI for the extensions, too

adriaandegroot committed:
CI: extract the tarball from calamares first

adriaandegroot committed:
Docs: mention the *mobile* module

adriaandegroot committed:
CI: massage message a bit and adjust naming scheme

adriaandegroot committed:
[image-slideshow] Add an example QML slideshow for images

The default slideshow is rather complicated for a "simple" slideshow
with images, so add an extensively-documented images-only slideshow
in QML. This can easily be extended with more images, or with
text and fanciness if the designer wants to flex more QML.

Commit:
[mobile] bugfix: initialise externalToInternal

This variable was uninitialised, leading to an undefined action occurring
when the install target screen is skipped.

adriaandegroot committed:
Merge pull request #8 from Undef-a/upstream/fs_selection

[mobile] Allow user to configure filesystem type

adriaandegroot committed:
CI: add the style tool from calamares

adriaandegroot committed:
[mobile] Apply coding style

adriaandegroot committed:
Changes: pre-release housekeeping

adriaandegroot committed:
CI: note that the release-script is Linux-only

adriaandegroot committed:
Docs: polish the README to match new additions

ollieparanoid committed:
Docs: README: update mobile description

adriaandegroot committed:
Import os-* modules from the main repo

The os-* branch was started to add some os-specific modules
to the main Calamares repo. Now that calamares-extensions
has its own release cycle and a reason-to-be beyond examples,
those modules are more suitable here than in the main repo.

Both modules are stubs; development did not get very far.

adriaandegroot committed:
Add os-* modules to the build

Document SKIP_MODULES and USE_os, although they don't **work**
yet; this should be the same machinery as in the core repo.

adriaandegroot committed:
CMake: bail out on broken CMake folders

On a developer's system, the CMake configuration from the **build**
can be found in cache, and that means that <srcdir>/build/CalamaresConfig.cmake
is read instead of an installed version; that doesn't work, so
bail out early (after a half-dozen errors from CalamaresConfig internals).

adriaandegroot committed:
[os-freebsd] Until there's a config file, mark it explicitly without

adriaandegroot committed:
[os-freebsd] Remove superfluous linking

For external modules, calamares_add_plugin() does the right thing,
and links to Calamares::calamares; don't use the internal name
from the Calamares repository, and don't link redundantly
to it anyway.

adriaandegroot committed:
CMake: collect and report the skipped modules

Use the newly-added methods for reporting skipped modules.
This requires a post-3.2.39 Calamares, which the CMakeLists.txt
does not yet enforce.

adriaandegroot committed:
Merge pull request #13 from ollieparanoid/readme-update

Docs: README: update mobile description

adriaandegroot committed:
[mobile] Don't need to list headers as source

adriaandegroot committed:
[unpackfsc] Stub of unpackfsc

This module is supposed to use fsarchiver (or similar tools)
to unpack a filesystem to the target. In many ways it is like
the rawfs module.

adriaandegroot committed:
[unpackfsc] Load and log configuration

adriaandegroot committed:
[unpackfsc] Simplify to **one** unpack action

The whole getup with multiple unpack actions makes the module
more complicated; instead, use one unpack action only. If there
are more of them, use multiple instances.

Similarly, remove the Config object: The **view** modules need a
Config object, because the config can be used by multiple UIs, and
we don't want to mix UI with business logic. For a single job,
that's different. Just put the job config in the job itself.

adriaandegroot committed:
[unpackfsc] Add stubs for calling the external tools.

This isn't the right way to do it: there is no possibility to
report progress or let the tool run asynchronously.

adriaandegroot committed:
CMake: Calamares is C++17, use it in extensions as well.

adriaandegroot committed:
CMake: Calamares is C++17, use it in extensions as well.

ollieparanoid committed:
[mobile] add option builtinVirtualKeyboard

Allow to always hide the built-in qtvirtualkeyboard to support using a
different keyboard.

Related: https://gitlab.com/postmarketOS/postmarketos-ondev/-/issues/47

ollieparanoid committed:
[mobile] add option userPasswordNumeric

Related: https://gitlab.com/postmarketOS/postmarketos-ondev/-/issues/46

ollieparanoid committed:
[mobile] rename default_pin -> user_pass

Prepare to add a config option to allow non-numeric passwords, by giving
the existing default_pin screen the more generic name user_pass.
Adjust the title of the screen too.

"default" in the file name was referring to configuring the default
user's PIN/password as opposed to the SSH user. However, I think
replacing it with "user" makes it more intuitive.

ollieparanoid committed:
[mobile] tweak invalid chars msg/related comments

Change "cannot be typed in at boot time" to "can possibly not be typed
in after installation" and explain that the same check is now used for
the user password too.

ollieparanoid committed:
[mobile] navNextFeature: fix skipping > 1 feature

Fix the condition in the while loop, so that it can actually continue
with the "continue" keyword. Without this patch, having the sshd
feature disabled would result in the fsType feature getting displayed,
even if it was disabled.

ollieparanoid committed:
mobile.qml: refactor navNextFeature()

Make it more readable as for-loop.

adriaandegroot committed:
CMake: bump version

adriaandegroot committed:
Docs: add release announcement

adriaandegroot committed:
CMake: update versioning infrastructure

Be smarter with versioning information: we don't
need to generate a whole build system to query
the version number set in CMake. Use script mode,
and print the version. Unfortunately, `project()`
isn't valid in script mode so we need to lift the
version into a variable and stick in some boilerplate
for printing the (full) version.

adriaandegroot committed:
Merge pull request #14 from ollieparanoid/non-numeric-pass

[mobile] support non-numeric passwords

adriaandegroot committed:
[mobile] Consistent initialization-expressions in declaration of Config

adriaandegroot committed:
CI: Use new versioning support from CMakeLists.txt

adriaandegroot committed:
CI: sign release tarball after it's made

- replace this manual step with signing automatically;
  the tag is signes as well so the gpg-agent likely
  still has the key.

adriaandegroot committed:
[unpackfsc] Start implementing fsarchive runner

adriaandegroot committed:
CMake: bump required Calamares version

We need the latest version with new failure-enum values.

adriaandegroot committed:
CI: switch to matrix notifications

adriaandegroot committed:
Merge branch 'update-release-reporting' into calamares

adriaandegroot committed:
[unpackfsc] Apply coding style

adriaandegroot committed:
[unpackfsc] Support functions for Runners

adriaandegroot committed:
[unpackfsc] Build a proxy for handling process output

- Rearrange sources, give implementation classes their own header
- Add a *RunCommand* class to run the commands for the tools.
  Existing code in System::runCommand() does not expose any
  of the output until the process is done. This means it's
  not useful for reporting progress information. New code
  **does** do that.

Rearrange sources, split to multiple

adriaandegroot committed:
[unpackfsc] Stub of the Unsquash runner

adriaandegroot committed:
Docs: mention the communication channels

adriaandegroot committed:
Docs: add a CONTRIBUTING document

- just copy the Calamares one and point to it for all technicalities

adriaandegroot committed:
Docs: badge for Matrix

adriaandegroot committed:
Merge branch 'add-unpackfsc' into calamares

adriaandegroot committed:
CI: repair the action YAML

adriaandegroot committed:
CI: restore the success/fail check, silence curl output

adriaandegroot committed:
CI: Add Matrix notification action

adriaandegroot committed:
CI: update to current matrix-notifications

adriaandegroot committed:
CI: try to use (shared) Calamares notification action

adriaandegroot committed:
CI: try to use (shared) Calamares notification action

adriaandegroot committed:
CI: switch to shared Calamares actions entirely

adriaandegroot committed:
CI: use generic-build action instead of expanding it here

adriaandegroot committed:
CI: factor out prepatation, chase update to build

adriaandegroot committed:
CI: chase new tag in actions repo

adriaandegroot committed:
Docs: IRC links to Libera.Chat

adriaandegroot committed:
Docs: do not use freenode

adriaandegroot committed:
CMake: adjust extensions to changes in Calamares core

adriaandegroot committed:
SPDX: add license texts

adriaandegroot committed:
[os-nixos] Replace useless stub implementation

The existing stub just did the same as slow-python, logging
some values and doing nothing useful. It didn't even act
as an example of the original os-nixos plan, which was
to write the config file.

Do a very minimal effort example with variable substitution
from GS.

adriaandegroot committed:
SPDX: wrangling

- remove / add license texts that are actually used
- the CMakeLists are supposed to be BSD-2-clause

adriaandegroot committed:
SPDX: add dep5-blanket for build artifacts and GH cruft

adriaandegroot committed:
SPDX: tag various

adriaandegroot committed:
SPDX: tag modules

adriaandegroot committed:
[os-nixos] Example global-config file, for tests

demmm committed:
[refind] adding a simple rEFInd module
takes just 2 options, install in /boot as EFI partition and
uses the refind-install option to install & create needed conf files
no work done to make EFI partiton configurable or choose a manual
option to install.  Those can be added, if there is any interest

demmm committed:
[refind] apply requested changes
tested on new ISO & installed with refind option

Otus9051 committed:
How is Kolkata in Europe?

adriaandegroot committed:
Merge pull request #16 from demmm/calamares

[refind] adding a simple rEFInd module

adriaandegroot committed:
[refind] kernel_params is a list (converted to space-separated later)

adriaandegroot committed:
[refind] Python stylings

- no need for explicit close() in file context
- simplify returns

adriaandegroot committed:
[refind] Improve device-detection warnings

- Possibly support something other that /boot
- Error message mismatch with what it actually checks

adriaandegroot committed:
[refind] Python stylings

- Use None for unset
- Use logging functions instead of print()

adriaandegroot committed:
[refind] Remove checks

- The checks use string manipulations and will always get
  a non-empty string, so the checks can't fail
- Refactor the check that there is an EFI partition mounted
- The refind-install command does not use any of the values
  computed by the checks.

adriaandegroot committed:
[refind] Add module to build

adriaandegroot committed:
CMake: enable tests

- needs to have loadmodule and test_conf in PATH, which
  means some tweaking is needed to actually *run* them.

adriaandegroot committed:
Merge pull request #17 from Otus9051/patch-1

[os-nixos] Test data - Kolkata is not in Europe

adriaandegroot committed:
[refind] Fix tests

- the file "refind.conf" was interpreted as a **Calamares** configuration
  file for the module; it isn't, it's for rEFInd, so rename it. The
  code does not use it yet, anyway.
- Mark the module as "noconfig" since it only consumes settings from
  the partition module and nothing else.

adriaandegroot committed:
CI: update to core calamares version of scripts

adriaandegroot committed:
[mobile] Keep the tests happy

adriaandegroot committed:
[filekeeper] Load part of the config file

adriaandegroot committed:
[refind] Fix warning function name

adriaandegroot committed:
[os-nixos] Avoid None-concatenate-with-str

When running tests, with no configurations, GS contains none
of the keys the Nix configuration is looking for. Avoid
TypeErrors (but still end up failing the test because we
can't run the nix-config tool at the end).

adriaandegroot committed:
CMake: bump version and Calamares-requirement

This is prep-work for a new release that uses the most
recent Calamares internals.

adriaandegroot committed:
[unpackfsc] Use libcalamares version of Runner now

adriaandegroot committed:
Remove filekeeper job

- the filekeeper was never fully implemented, while core Calamares
  has a preservefiles module that does do all the same things.

adriaandegroot committed:
[unpackfsc] Factor out command-name, it doesn't need translation

adriaandegroot committed:
[unpackfsc] Implement most of unsquash

adriaandegroot committed:
[unpackfsc] Hook up progress

adriaandegroot committed:
[unpackfsc] Remove unreachable progress

adriaandegroot committed:
[unpackfs] Include a message in progress reports

adriaandegroot committed:
[unpackfsc] Map destination path to target system

adriaandegroot committed:
[unpackfsc] Get the total number of inodes

- Call run() to actually run the command we've set up; in the
  advice "have you tried turning if off and on again", the "on"
  bit is actually kind of important.

adriaandegroot committed:
[unpackfs] Don't reformat table of enum names

adriaandegroot committed:
[unpackfsc] Avoid newlines in filenames

adriaandegroot committed:
[unpackfsc] strip() is Python, I guess

adriaandegroot committed:
[unpackfsc] Report progress through status message

adriaandegroot committed:
[unpackfsc] Fix build with old Qt

adriaandegroot committed:
CI: fix build-script with new-style version reporting

adriaandegroot committed:
Changes: post-release housekeeping

adriaandegroot committed:
[unpackfsc] Report fsarchiver progress, too

adriaandegroot committed:
[unpackfsc] Add test-configs for fsarchiver-unpack

adriaandegroot committed:
[unpackfsc] Use restdir instead of restfs

adriaandegroot committed:
[unpackfsc] Reduce number of status updates in fsarchiver

adriaandegroot committed:
[unpackfs] Document that fsarchiver uses savedir/restdir

adriaandegroot committed:
Merge branch 'fsarchiver' into calamares

adriaandegroot committed:
Changes: pre-release housekeeping

vlinkz committed:
Add NixOS modules and remove others

vlinkz committed:
Update nixos-customize-config to add keyboard selection

vlinkz committed:
Update modules and refactor config generation

vlinkz committed:
EFI grub device always nodev

vlinkz committed:
Truncate nixos-version

vlinkz committed:
Remove redundent services

vlinkz committed:
Remove broken swap options

vlinkz committed:
Format and add some checks for settings groups

vlinkz committed:
Fix luks and swap

Fixed luks encryption
Mount swap during configuration
Extend slideshow timing
Change title to remove calamares name

vlinkz committed:
Add support for multiple luks encrypted partitions

vlinkz committed:
Add desktop selection

vlinkz committed:
Use nmapplet option
Change from using nmapplet package to options
Switch producticon used in branding

vlinkz committed:
Merge modules into singular  module

vlinkz committed:
Switch efi to systemd-boot

vlinkz committed:
Fix LUKS and swap

vlinkz committed:
Remove lumina desktop and clean up

Not yet stable enough as per https://github.com/NixOS/nixpkgs/issues/168532

vlinkz committed:
Remove unfree packages and run as user

vlinkz committed:
Allow selection of unfree packages

vlinkz committed:
Add descriptions
- Adds descriptions to desktop environments
- Reorder config steps

vlinkz committed:
Switch to libcalamares calls
Use calls from libcalamares rather than subprocess in more cases
Also adds more status messages while installing

vlinkz committed:
Fix error messages

vlinkz committed:
Misc fixes
- Fixed boot.extraModulePackages = [ ] with two spaces when items are removed
- Fixed autologin getty being set when user didn't pick autologin

vlinkz committed:
Actually fix whitespace and format

vlinkz committed:
Update desktop descriptions

vlinkz committed:
Fix console keymap and remove unecessary packages

vlinkz committed:
Fix partitions and luks

vlinkz committed:
Fix tty autologin and misc options
- Fix tty autologin
- Remove cups and sound for No Desktop systems

vlinkz committed:
Disable networking for No Desktop

vlinkz committed:
Switch ping url and add locale geoip

vlinkz committed:
Add licensing to config files

vlinkz committed:
Add some branding licensing

vlinkz committed:
Image licensing

vlinkz committed:
Re-add networkmanager for No Desktop choice

vlinkz committed:
Remove typo in config

vlinkz committed:
Improve installation logging

vlinkz committed:
Add some packages to user rather than system

vlinkz committed:
Add cache.nixos.org internet check url

vlinkz committed:
Support locales from glibc SUPPORTED file

Commit:
Add Budgie as a desktop option

Signed-off-by: Federico Damián Schonborn <fdschonborn@gmail.com>

vlinkz committed:
Merge pull request #18 from FedericoSchonborn/budgie

Add Budgie as an option on the Desktop section

vlinkz committed:
Use the default efi mount point `/boot`

vlinkz committed:
Merge pull request #19 from NixOS/defaultefimount

Use the default efi mount point `/boot`

wineee committed:
Add Deepin as an option on the Desktop section

vlinkz committed:
Fix LUKS keyfile exposure

vlinkz committed:
Merge pull request #21 from NixOS/luksfix

Fix LUKS keyfile exposure

vlinkz committed:
Merge pull request #20 from wineee/deepin

Add Deepin as an option on the Desktop section

ElvishJerricco committed:
Revert "Fix LUKS keyfile exposure"

This reverts commit 6228cd3e1ac1d095fb1ade1536cff1d51e9f1900.

ElvishJerricco committed:
Do not use crypto_keyfile.bin in UEFI, but leave BIOS the same.

vlinkz committed:
Merge pull request #25 from ElvishJerricco/bios-use-crypto-keyfile

Do not use crypto_keyfile.bin in UEFI, but leave BIOS the same.

vlinkz committed:
Updates for 3.3.0 and luks2

ibbem committed:
Use the Firefox module instead of the package directly

The Firefox module installs and enables the selected locale whereas the
Firefox package just uses the default locale. Hence, Firefox's language
was still English even when choosing a language other than English.

FriedrichAltheide committed:
Use proxy when installing, if set

vlinkz committed:
Merge pull request #26 from vlinkz/3.3.0

Updates for 3.3.0 and luks2

vlinkz committed:
Merge pull request #28 from ibbem/use-firefox-module

Use the Firefox module instead of the package directly

vlinkz committed:
Define deafult partition layout

eclairevoyant committed:
fix pipewire sound config

vlinkz committed:
Merge pull request #30 from eclairevoyant/fix-sound-config

fix pipewire sound config

K900 committed:
Advertise Plasma 6, adjust config for Plasma 5

RaitoBezarius committed:
Merge pull request #33 from K900/plasma-6

Advertise Plasma 6, adjust config for Plasma 5

vlinkz committed:
Don't mount swap twice

vlinkz committed:
Update deprecated xkb options

vlinkz committed:
Merge pull request #36 from vlinkz/swapfix

Don't mount swap twice

vlinkz committed:
fix crypto_keyfile creation condition

ElvishJerricco committed:
showNotEncryptedBootMessage: false

This is no longer needed, now that we correctly detect whether a
crypto_keyfile.bin is required in `/boot`.

ElvishJerricco committed:
Merge pull request #43 from vlinkz/cryptofix

Fix `crypto_keyfile.bin` creation condition

DerRockWolf committed:
fix(mount): boot umask

ElvishJerricco committed:
Merge pull request #46 from DerRockWolf/fix-mount-boot-umask

fix(mount): EFI umask

lulu-berlin committed:
Bootstrap testing

- Add a unit test for the baseline configuration
- All mocks are extracted into `conftest.py`
- Add fixture files for `hardware-configuration.nix` and `kbd-model-map`
- Add a nix flake for running and developing the tests
- Add instructions to README.md
- Configure github actions to run tests on all pull requests

lulu-berlin committed:
Trivial linting

- Autoformat the code with `black` for better readability
- Replace `== True` with `is True`
- Remove redundant parentheses in conditional statements
- Replace `if not x in y:` with `if x not in y:`
- Replace `!= None` with `is not None`
- Use raw string (r"...") with regular expressions to avoid escaping
  issues (as recommended in the documentation of the `re` module).

vlinkz committed:
Merge pull request #48 from lulu-berlin/linting

Trivial linting

vlinkz committed:
Merge pull request #29 from FriedrichAltheide/proxy-settings

Use proxy when installing, if proxy is set

vlinkz committed:
Merge pull request #47 from lulu-berlin/testing

Bootstrap testing

bjornfor committed:
Don't emit autoLogin config with eval warning

Fix this evaluation warning on the first rebuild after install:

  [test@nixos:~]$ sudo nixos-rebuild build
  building Nix...
  building the system configuration...
  evaluation warning: The option `services.xserver.displayManager.autoLogin' defined in `/etc/nixos/configuration.nix' has been renamed to `services.displayManager.autoLogin'.

vlinkz committed:
Merge pull request #49 from bjornfor/dont-emit-config-with-eval-warning

Don't emit autoLogin config with eval warning

ElvishJerricco committed:
Respect Kernel setting in /etc/nixos-generate-config.conf

vlinkz committed:
Merge pull request #51 from ElvishJerricco/push-xomprrmsksrq

Respect Kernel setting in /etc/nixos-generate-config.conf

K900 committed:
Drop Plasma 5

It's deprecated in 25.05, so we should stop recommending it for new installs.

ElvishJerricco committed:
Merge pull request #52 from K900/o7

Drop Plasma 5

CoffeeImpliesCode committed:
Change default /boot partition size to 1GiB

ElvishJerricco committed:
Merge pull request #53 from CoffeeImpliesCode/calamares

Change default /boot partition size to 1GiB

K900 committed:
Update Pulseaudio option name

ElvishJerricco committed:
Merge pull request #58 from K900/fix-warning

Update Pulseaudio option name

K900 committed:
calamares: 3.3.13 -> 3.4.0, qt6-ize, clean up, move to by-name

K900 committed:
calamares-nixos-extensions: update for vendored sources + Calamares 3.4.0

K900 commented:
Strictly speaking, we're probably fine keeping this GPL, but it would be nice to get signoffs from everyone involved to just relicense as MIT and not think about it. So:

- [ ] @vlinkz
- [ ] @FedericoSchonborn 
- [ ] @wineee 
- [ ] @ElvishJerricco 
- [ ] @eclairevoyant 
- [ ] @lulu-berlin 
- [ ] @FriedrichAltheide 
- [ ] @bjornfor

The slideshow QML is technically GPLv3 as well, but it is also absolutely trivial so I don't think that's an issue.

K900 committed:
nixos/partition-manager: switch to qt6

K900 committed:
nixos/iso: enable partition-manager (for kpmcore)

eclairevoyant commented:
> it would be nice to get signoffs from everyone involved to just relicense as MIT

Sure.

K900 commented:
@bjornfor can you please add an explicit comment agreeing to the relicense, so we have a paper trail?

bjornfor commented:
I agree to re-license to MIT.

FedericoSchonborn commented:
Hi, I agree

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/install-stuck-at-46-still-an-issue/67596/6

K900 committed:
calamares-nixos-extensions: vendor sources

K900 committed:
calamares-nixos: convert to wrapper

lulu-berlin commented:
> > Strictly speaking, we're probably fine keeping this GPL, but it would be nice to get signoffs from everyone involved to just relicense as MIT and not think about it. So:
> > 
> > (please comment on the issue in addition to checking the box, just so everything is very clear)
> > 
> > * [ ]  @vlinkz
> > * [x]  @FedericoSchonborn
> > * [x]  @wineee
> > * [ ]  @ElvishJerricco
> > * [x]  @eclairevoyant
> > * [x]  @lulu-berlin
> > * [ ]  @FriedrichAltheide
> > * [x]  @bjornfor
> > 
> > The slideshow QML is technically GPLv3 as well, but it is also absolutely trivial so I don't think that's an issue.
> 
> I agree for my code

I agree.

FriedrichAltheide commented:
> Honestly the entire config generation big can (and should) be redone, and we can easily make it way more flexible than what you're proposing.

I'm curious to see what this flexibility will look like. Please consider backporting the idea of https://github.com/NixOS/calamares-nixos-extensions/pull/40/commits/f7a63a3362bccc93ad234762deb3e87562dfe3db

ElvishJerricco reviewed (commented):
Other than the patch thing, LGTM.

emilazy commented on nixos/modules/installer/cd-dvd/installation-cd-graphical-calamares-plasma5.nix:
```diff
@@ -45,8 +45,8 @@
       }
-      ln -sfT ${pkgs.calamares-nixos}/share/applications/io.calamares.calamares.desktop ${
-        desktopDir + "io.calamares.calamares.desktop"
+      ln -sfT ${pkgs.calamares-nixos}/share/applications/calamares.desktop ${
+        desktopDir + "calamares.desktop"
```
Is this a deliberate upstream change? It seems odd for them to move away from the increasingly‐common reversed DNS format.

Please squash these changes into another commit to reduce the number of broken in‐between states.

emilazy replied:
Okay, acknowledged. Though I wonder if there was any particular reason we were doing so.

emilazy reviewed (changes requested):
Thank you for this!

The vendoring merge is incorrect: the second parent of f08cde4ef6e88585aa8fcf1aa0d7c10307a5a9dd is 4ca31c1a6e3ac4d5b7733fc64662f7eb9c7754aa, which does not match the hash of https://github.com/NixOS/calamares-nixos-extensions/commit/63620df329c29fafc4928414fcde5d73f5b24f31. I think you accidentally rewrote all the commits; please fix that.

Otherwise, looks mostly good to me.

K900 commented on pkgs/by-name/ca/calamares/package.nix:
```diff
@@ -0,0 +1,123 @@
+  ];
+
+  postPatch = ''
+    # this is called via pkexec, which does not resolve symlinks, so the policy
+    # needs to point at the symlinked path
```
Yes.

K900 commented on nixos/modules/programs/partition-manager.nix:
```diff
@@ -15,7 +15,7 @@ in
       enable = lib.mkEnableOption "KDE Partition Manager";
 
-      package = lib.mkPackageOption pkgs [ "libsForQt5" "partitionmanager" ] { };
+      package = lib.mkPackageOption pkgs [ "kdePackages" "partitionmanager" ] { };
```
No, the release note will be "we deleted all of Plasma 5".

K900 commented on pkgs/by-name/ca/calamares/global.patch:
```diff
@@ -0,0 +1,33 @@
+diff --git a/src/libcalamares/GlobalStorage.h b/src/libcalamares/GlobalStorage.h
```
Discussing on Matrix now.

K900 commented on pkgs/by-name/ca/calamares-nixos/package.nix:
```diff
@@ -0,0 +1,30 @@
+    makeWrapper ${lib.getExe calamares} $out/bin/calamares \
+      --prefix XDG_DATA_DIRS : ${calamares-nixos-extensions}/share \
+      --prefix XDG_CONFIG_DIRS : ${calamares-nixos-extensions}/etc \
+      --add-flag -X
```
Done.

K900 replied:
Done.

K900 commented on pkgs/by-name/ca/calamares-nixos-extensions/src/testing/test_baseline.py:
```diff
@@ -1,153 +0,0 @@
-import subprocess
```
It didn't trivially run so I kinda gave up.

vlinkz commented:
> Strictly speaking, we're probably fine keeping this GPL, but it would be nice to get signoffs from everyone involved to just relicense as MIT and not think about it. So:
> 
> (please comment on the issue in addition to checking the box, just so everything is very clear)
> 
> * [x]  @vlinkz
> * [x]  @FedericoSchonborn
> * [x]  @wineee
> * [x]  @ElvishJerricco
> * [x]  @eclairevoyant
> * [x]  @lulu-berlin
> * [x]  @FriedrichAltheide
> * [x]  @bjornfor
> 
> The slideshow QML is technically GPLv3 as well, but it is also absolutely trivial so I don't think that's an issue.

I agree

K900 commented:
Thanks!

K900 committed:
calamares-nixos-extensions: remove more outdated docs, update license

vlinkz reviewed (changes requested):
Testing this out noticed a couple of regressions
- Removing 0007 breaks setting user keyboard on GNOME, but this should be fixed upstream by GNOME so we can just wait instead of patching https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7761
- Removing 0003 and using pkexec directly breaks opening links in Firefox, as it tries to open them as root: `Running Firefox as root in a regular user's session is not supported.  ($XAUTHORITY is /run/user/1000/.mutter-Xwaylandauth.30BWA3 which is owned by nixos.)`. We could wrap xdg-open for calamares to open as user instead.

K900 commented:
0007 is fixed upstream: https://codeberg.org/Calamares/calamares/commit/fb5af7d5da4e5e4e92a2718be0e140f3e856dd45

0003 is weird, do other distributions just not have links in the installer? Maybe we should just remove the links then?

vlinkz commented:
> 0007 is fixed upstream: https://codeberg.org/Calamares/calamares/commit/fb5af7d5da4e5e4e92a2718be0e140f3e856dd45

Looks like we need to enable the fix in [`keyboard.conf`](https://github.com/NixOS/nixpkgs/pull/430636/files#diff-02d352568d7c34476506bf46e27672abe79aa3681c26a1d0d7d92b45f4cfae79) though https://codeberg.org/Calamares/calamares/src/commit/fb5af7d5da4e5e4e92a2718be0e140f3e856dd45/src/modules/keyboard/keyboard.conf#L47

vlinkz commented:
> 0003 is weird, do other distributions just not have links in the installer? Maybe we should just remove the links then?

Tested EndeavourOS, Neon, and Manjaro. All have the same issues and just remove the links 💀

K900 committed:
calamares-nixos-extensions: clean up config files

- remove inline comments from upstream, link to upstream files instead
- remove settings that match upstream defaults
- reformat

K900 committed:
calamares-nixos-extensions: use recommended password quality settings

K900 committed:
calamares-nixos-extensions: bump required RAM

There's really no chance of 1GB of RAM being able to eval a NixOS config.

K900 commented:
OK, pushed config change for 0007, and a very stupid workaround for 0003 (that nonetheless seems to work), along with a bunch of config cleanups/updates.

K900 committed:
calamares: drop privileges before xdg-open

vlinkz reviewed (approved):
LGTM. Only thing not working now is keyboard changes on plasma6, but that's a harder issue to solve and probably needs better upstream support.
- locale1 doesn't work because `Failed to set keymap: Changing system settings via systemd is not supported on NixOS.`
- calamares kwin doesn't work because the kwin keyboard apply function checks root dirs (`/root/.config`) (and regardless `~/.config/kxkbrc` doesn't exist by default)

ElvishJerricco commented:
What's up with the green/red/yellow icons?
<img width="1564" height="1086" alt="bad-user-pass" src="https://github.com/user-attachments/assets/43f3854f-3dff-443b-bba4-0a7b80ccdf03" />
They used to look like this. Notice the icons that were gone on the username, and of course the warning and error sign icons.
<img width="1564" height="1086" alt="good-user-pass" src="https://github.com/user-attachments/assets/875fc205-b445-4888-aefa-c084f85d8745" />
Also the allow unfree screen lost its styling, but I dunno if we care.
<img width="1564" height="1086" alt="bad-allow-unfree" src="https://github.com/user-attachments/assets/6c2ce23c-4545-42ce-97d1-bded9f537003" />
Before:
<img width="1564" height="1086" alt="good-allow-unfree" src="https://github.com/user-attachments/assets/4d825f73-9363-4438-a007-2b7f6bc33a45" />

K900 commented:
The icons are a qtsvg bug, there should be a fix in 6.9.2. The unfree page I rewrote entirely, it can definitely be made to look nicer.

ElvishJerricco reviewed (approved):
As long as we don't think those icons are a problem, LGTM

emilazy reviewed (approved):
I hate the browser `pkexec` thing but otherwise this seems good.

philiptaron commented:
This PR causes the size of the nixpkgs tarball to grow by 10MB.

```
$ gh api /repos/nixos/nixpkgs/tarball/$(git rev-parse 8baeaa3e949e52844b9cfda86af3c8cc1829dcd5^) | wc -c | numfmt --to=iec --suffix=B
44MB
$ gh api /repos/nixos/nixpkgs/tarball/$(git rev-parse 8baeaa3e949e52844b9cfda86af3c8cc1829dcd5) | wc -c | numfmt --to=iec --suffix=B
54MB
```

K900 commented:
It's the assets, presumably. We might want to JPEG those.
